Feature/list dict

screameryo opened the pull request.

screameryo committed:
Обновление проектов и генерация тестовых данных

Удален контент из List.xml.
Изменены ссылки на проекты в Bank.sln.
Обновлены версии сборок в .csproj файлах.
Добавлена зависимость от Bogus в BankSystem.App.csproj.
Удалены старые примеры и добавлен новый код в Program.cs.
Добавлен класс TestDataGenerator для генерации фейковых данных.

screameryo committed:
Обновление версий и изменение метода поиска клиента

В файле Program.cs изменен метод поиска клиента:
- Удален поиск с использованием FirstOrDefault
- Добавлен поиск с использованием LastOrDefault
- Добавлен поиск клиента по ключу в словаре

PetrovskiAlex commented on BankSystem.App/Services/TestDataGenerator.cs:
```diff
@@ -0,0 +1,71 @@
+{
+    public class TestDataGenerator
+    {
+        public void GenerateEmployee(ref List<Employee> employees)
```
использование ref нужно в очень редких случаях

это не тот случай

PetrovskiAlex commented on BankSystem.App/Services/TestDataGenerator.cs:
```diff
@@ -0,0 +1,71 @@
+                .RuleFor(e => e.Department, f => f.Name.JobArea())
+                .RuleFor(e => e.Contract, f => f.Random.Bool() ? $"Contract to {DateTime.Now.AddMonths(f.Random.Number(1, 24)).ToString("d")}" : null);
+
+            for (int i = 0; i < 1000; i++)
```
у Bogus'a есть метод, который умеет делать список

PetrovskiAlex commented on BankSystem.App/Services/TestDataGenerator.cs:
```diff
@@ -0,0 +1,71 @@
+            }
+        }
+
+        public void GenerateClient(ref List<Client> clients)
```
те же самые комментарии что и выше

PetrovskiAlex commented on BankSystem.App/Services/TestDataGenerator.cs:
```diff
@@ -0,0 +1,71 @@
+                {
+                    clients.Add(client.Telephone, client);
+                }
+                catch (Exception)
```
так как у нас тут нереальные данные и нам всё-равно какие они там, придумайте что-то чтобы кода было меньше, без обработки исключений

PetrovskiAlex commented on Practice/Program.cs:
```diff
@@ -1,62 +1,105 @@
             var newemployee = bankService.ConvertClientToEmployee(client, "Backend", "Develop", 100000);
             Console.WriteLine($"Client now is jbject employee {Environment.NewLine}");
+
+            var testDataGenerator = new TestDataGenerator();
```
Разбили бы как-то этот большой метод Main, а то лежит всё вместе одним полотном

screameryo committed:
Изменены методы генерации данных в `TestDataGenerator.cs`:
- `GenerateEmployee` теперь возвращает список сотрудников
- `GenerateClient` теперь возвращает список клиентов
- `GenerateClientAsDictionary` теперь возвращает словарь клиентов по паспорту

В `Program.cs` добавлены новые методы и изменена логика:
- Добавлены методы `ChangeContract` и `ChangeCurrency`
- Добавлен метод `PracticeOneRefValType` для демонстрации работы с типами
- Добавлен метод `PracticeTwoListDict` для использования обновленных методов генерации данных и работы со списками и словарями
